pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -294,6 +294,7 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 	ch := utils.NewComponentHandler(log, r.Client, r.Scheme, managementClusterConnection)
 	guardianCfg := &render.GuardianConfiguration{
 		URL:               managementClusterConnection.Spec.ManagementClusterAddr,
+		TunnelCAType:      managementClusterConnection.Spec.TLS.CA,
```

Comment:
We usually write back defaults to the CRD from within the controllers if values are not provided. If you check for "filldefaults" in this repo, you will find some examples.